File: src/SaxpyLoop.java
public class SaxpyLoop {
   public static void main(String []args) {
	   int N = 1 << 26;
	   float XVAL = (float)(10 * Math.random());
	   float YVAL = (float)(10 * Math.random());
	   float AVAL = (float)(10 * Math.random());
	   float[] x = new float[N];
	   float[] y = new float[N];

	   System.out.println("N: " + N);
	   
	   for (int i=0; i<N; i++) {
		   x[i] = XVAL;
		   y[i] = YVAL;
	   }
	   
	   double startTime = System.nanoTime();
	   for (int i=0; i<N; i++) {
		   y[i] += AVAL * x[i];
	   }
	   double endTime = System.nanoTime();
	   System.out.println("Elapsed: " + (endTime - startTime) / 1000000.0 + " ms");
	   
	   float error = 0;
	   for (int i=0; i<N; i++) {
		   error += Math.abs(y[i] - (YVAL + AVAL * XVAL));
	   }
	   
	   System.out.println("Error: " + error);
   }
}

File: src/saxpy_loop.R
N <- 2 ^ 26

# Random numbers
XVAL <- 10 * runif(1)
YVAL <- 10 * runif(1)
AVAL <- 10 * runif(1)

x <- array(XVAL, dim=c(N))
y <- array(YVAL, dim=c(N))
cat("N:", N, "\n")

t0 <- proc.time()
for(i in 1:N) {
  y[i] <- y[i] + x[i] * AVAL
}
t1 = proc.time()
diff <- t1 - t0
cat("Elapsed:", diff*1000, " ms\n")

answer <- YVAL + AVAL * XVAL
error <- sum(abs(y - answer))
cat("Error:", error, "\n")
